### ISSUE TITLE
Editor segfaults in PropertyValueEvaluator::edit when closed

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#9653]
**Operating system or device - Godot version:**
Mageia 6 x86_64 - 2.1 branch cf52c63

**Issue description:**
When closing the editor, it segfaults with this backtrace:
```
Thread 1 "godot-2.1" received signal SIGSEGV, Segmentation fault.
0x0000000000d2343a in PropertyValueEvaluator::edit (this=0xeaeaeaeaeaeaeaea, p_obj=0x0) at editor/property_editor.cpp:4307
warning: Source file is more recent than executable.
4307            obj = p_obj;
(gdb) bt
#0  0x0000000000d2343a in PropertyValueEvaluator::edit (this=0xeaeaeaeaeaeaeaea, p_obj=0x0) at editor/property_editor.cpp:4307
#1  0x0000000000d1c901 in PropertyEditor::edit (this=0x72b1bc0, p_object=0x0) at editor/property_editor.cpp:3625
#2  0x0000000000c284c8 in ScriptEditorDebugger::_clear_remote_objects (this=0x729e050) at editor/script_editor_debugger.cpp:1678
#3  0x0000000000c2e0d2 in ScriptEditorDebugger::~ScriptEditorDebugger (this=0x729e050, __in_chrg=<optimized out>) at editor/script_editor_debugger.cpp:2031
#4  0x000000000078202f in memdelete<Node> (p_class=0x729e050) at core/os/memory.h:279
#5  0x0000000000dd04b6 in Node::_notification (this=0x5b03850, p_notification=1) at scene/main/node.cpp:131
#6  0x000000000045c4c4 in Node::_notificationv (this=0x5b03850, p_notification=1, p_reversed=true) at ./scene/main/node.h:44
#7  0x000000000045cac7 in CanvasItem::_notificationv (this=0x5b03850, p_notification=1, p_reversed=true) at ./scene/2d/canvas_item.h:87
#8  0x000000000045d09f in Control::_notificationv (this=0x5b03850, p_notification=1, p_reversed=true) at ./scene/gui/control.h:50
#9  0x0000000000565fe3 in Container::_notificationv (this=0x5b03850, p_notification=1, p_reversed=true) at ./scene/gui/container.h:37
#10 0x00000000005665bb in BoxContainer::_notificationv (this=0x5b03850, p_notification=1, p_reversed=true) at ./scene/gui/box_container.h:37
#11 0x00000000005674df in VBoxContainer::_notificationv (this=0x5b03850, p_notification=1, p_reversed=true) at ./scene/gui/box_container.h:80
#12 0x00000000017a861a in Object::notification (this=0x5b03850, p_notification=1, p_reversed=true) at core/object.cpp:863
#13 0x00000000017a609f in Object::_predelete (this=0x5b03850) at core/object.cpp:259
#14 0x00000000017b0648 in predelete_handler (p_object=0x5b03850) at core/object.cpp:1781
#15 0x000000000078200f in memdelete<Node> (p_class=0x5b03850) at core/os/memory.h:277
#16 0x0000000000dd04b6 in Node::_notification (this=0x5b02c70, p_notification=1) at scene/main/node.cpp:131
#17 0x000000000045c4c4 in Node::_notificationv (this=0x5b02c70, p_notification=1, p_reversed=true) at ./scene/main/node.h:44
#18 0x000000000045cac7 in CanvasItem::_notificationv (this=0x5b02c70, p_notification=1, p_reversed=true) at ./scene/2d/canvas_item.h:87
#19 0x000000000045d09f in Control::_notificationv (this=0x5b02c70, p_notification=1, p_reversed=true) at ./scene/gui/control.h:50
#20 0x0000000000565fe3 in Container::_notificationv (this=0x5b02c70, p_notification=1, p_reversed=true) at ./scene/gui/container.h:37
#21 0x0000000000adbffb in PanelContainer::_notificationv (this=0x5b02c70, p_notification=1, p_reversed=true) at ./scene/gui/panel_container.h:37
#22 0x00000000017a861a in Object::notification (this=0x5b02c70, p_notification=1, p_reversed=true) at core/object.cpp:863
#23 0x00000000017a609f in Object::_predelete (this=0x5b02c70) at core/object.cpp:259
#24 0x00000000017b0648 in predelete_handler (p_object=0x5b02c70) at core/object.cpp:1781
#25 0x000000000078200f in memdelete<Node> (p_class=0x5b02c70) at core/os/memory.h:277
#26 0x0000000000dd04b6 in Node::_notification (this=0x5551690, p_notification=1) at scene/main/node.cpp:131
#27 0x000000000045c4c4 in Node::_notificationv (this=0x5551690, p_notification=1, p_reversed=true) at ./scene/main/node.h:44
#28 0x000000000045cac7 in CanvasItem::_notificationv (this=0x5551690, p_notification=1, p_reversed=true) at ./scene/2d/canvas_item.h:87
#29 0x000000000045d09f in Control::_notificationv (this=0x5551690, p_notification=1, p_reversed=true) at ./scene/gui/control.h:50
---Type <return> to continue, or q <return> to quit---
#30 0x0000000000565fe3 in Container::_notificationv (this=0x5551690, p_notification=1, p_reversed=true) at ./scene/gui/container.h:37
#31 0x00000000008126e1 in SplitContainer::_notificationv (this=0x5551690, p_notification=1, p_reversed=true) at ./scene/gui/split_container.h:37
#32 0x0000000000813597 in VSplitContainer::_notificationv (this=0x5551690, p_notification=1, p_reversed=true) at ./scene/gui/split_container.h:96
#33 0x00000000017a861a in Object::notification (this=0x5551690, p_notification=1, p_reversed=true) at core/object.cpp:863
#34 0x00000000017a609f in Object::_predelete (this=0x5551690) at core/object.cpp:259
#35 0x00000000017b0648 in predelete_handler (p_object=0x5551690) at core/object.cpp:1781
#36 0x000000000078200f in memdelete<Node> (p_class=0x5551690) at core/os/memory.h:277
#37 0x0000000000dd04b6 in Node::_notification (this=0x5550990, p_notification=1) at scene/main/node.cpp:131
#38 0x000000000045c4c4 in Node::_notificationv (this=0x5550990, p_notification=1, p_reversed=true) at ./scene/main/node.h:44
#39 0x000000000045cac7 in CanvasItem::_notificationv (this=0x5550990, p_notification=1, p_reversed=true) at ./scene/2d/canvas_item.h:87
#40 0x000000000045d09f in Control::_notificationv (this=0x5550990, p_notification=1, p_reversed=true) at ./scene/gui/control.h:50
#41 0x0000000000565fe3 in Container::_notificationv (this=0x5550990, p_notification=1, p_reversed=true) at ./scene/gui/container.h:37
#42 0x00000000005665bb in BoxContainer::_notificationv (this=0x5550990, p_notification=1, p_reversed=true) at ./scene/gui/box_container.h:37
#43 0x00000000005674df in VBoxContainer::_notificationv (this=0x5550990, p_notification=1, p_reversed=true) at ./scene/gui/box_container.h:80
#44 0x00000000017a861a in Object::notification (this=0x5550990, p_notification=1, p_reversed=true) at core/object.cpp:863
#45 0x00000000017a609f in Object::_predelete (this=0x5550990) at core/object.cpp:259
#46 0x00000000017b0648 in predelete_handler (p_object=0x5550990) at core/object.cpp:1781
#47 0x000000000078200f in memdelete<Node> (p_class=0x5550990) at core/os/memory.h:277
#48 0x0000000000dd04b6 in Node::_notification (this=0x554fc70, p_notification=1) at scene/main/node.cpp:131
#49 0x000000000045c4c4 in Node::_notificationv (this=0x554fc70, p_notification=1, p_reversed=true) at ./scene/main/node.h:44
#50 0x000000000045cac7 in CanvasItem::_notificationv (this=0x554fc70, p_notification=1, p_reversed=true) at ./scene/2d/canvas_item.h:87
#51 0x000000000045d09f in Control::_notificationv (this=0x554fc70, p_notification=1, p_reversed=true) at ./scene/gui/control.h:50
#52 0x0000000000565fe3 in Container::_notificationv (this=0x554fc70, p_notification=1, p_reversed=true) at ./scene/gui/container.h:37
#53 0x00000000008126e1 in SplitContainer::_notificationv (this=0x554fc70, p_notification=1, p_reversed=true) at ./scene/gui/split_container.h:37
#54 0x0000000000812dad in HSplitContainer::_notificationv (this=0x554fc70, p_notification=1, p_reversed=true) at ./scene/gui/split_container.h:87
#55 0x00000000017a861a in Object::notification (this=0x554fc70, p_notification=1, p_reversed=true) at core/object.cpp:863
#56 0x00000000017a609f in Object::_predelete (this=0x554fc70) at core/object.cpp:259
#57 0x00000000017b0648 in predelete_handler (p_object=0x554fc70) at core/object.cpp:1781
#58 0x000000000078200f in memdelete<Node> (p_class=0x554fc70) at core/os/memory.h:277
#59 0x0000000000dd04b6 in Node::_notification (this=0x554c8b0, p_notification=1) at scene/main/node.cpp:131
---Type <return> to continue, or q <return> to quit---
#60 0x000000000045c4c4 in Node::_notificationv (this=0x554c8b0, p_notification=1, p_reversed=true) at ./scene/main/node.h:44
#61 0x000000000045cac7 in CanvasItem::_notificationv (this=0x554c8b0, p_notification=1, p_reversed=true) at ./scene/2d/canvas_item.h:87
#62 0x000000000045d09f in Control::_notificationv (this=0x554c8b0, p_notification=1, p_reversed=true) at ./scene/gui/control.h:50
#63 0x0000000000565fe3 in Container::_notificationv (this=0x554c8b0, p_notification=1, p_reversed=true) at ./scene/gui/container.h:37
#64 0x00000000008126e1 in SplitContainer::_notificationv (this=0x554c8b0, p_notification=1, p_reversed=true) at ./scene/gui/split_container.h:37
#65 0x0000000000812dad in HSplitContainer::_notificationv (this=0x554c8b0, p_notification=1, p_reversed=true) at ./scene/gui/split_container.h:87
#66 0x00000000017a861a in Object::notification (this=0x554c8b0, p_notification=1, p_reversed=true) at core/object.cpp:863
#67 0x00000000017a609f in Object::_predelete (this=0x554c8b0) at core/object.cpp:259
#68 0x00000000017b0648 in predelete_handler (p_object=0x554c8b0) at core/object.cpp:1781
#69 0x000000000078200f in memdelete<Node> (p_class=0x554c8b0) at core/os/memory.h:277
#70 0x0000000000dd04b6 in Node::_notification (this=0x5549540, p_notification=1) at scene/main/node.cpp:131
#71 0x000000000045c4c4 in Node::_notificationv (this=0x5549540, p_notification=1, p_reversed=true) at ./scene/main/node.h:44
#72 0x000000000045cac7 in CanvasItem::_notificationv (this=0x5549540, p_notification=1, p_reversed=true) at ./scene/2d/canvas_item.h:87
#73 0x000000000045d09f in Control::_notificationv (this=0x5549540, p_notification=1, p_reversed=true) at ./scene/gui/control.h:50
#74 0x0000000000565fe3 in Container::_notificationv (this=0x5549540, p_notification=1, p_reversed=true) at ./scene/gui/container.h:37
#75 0x00000000008126e1 in SplitContainer::_notificationv (this=0x5549540, p_notification=1, p_reversed=true) at ./scene/gui/split_container.h:37
#76 0x0000000000812dad in HSplitContainer::_notificationv (this=0x5549540, p_notification=1, p_reversed=true) at ./scene/gui/split_container.h:87
#77 0x00000000017a861a in Object::notification (this=0x5549540, p_notification=1, p_reversed=true) at core/object.cpp:863
#78 0x00000000017a609f in Object::_predelete (this=0x5549540) at core/object.cpp:259
#79 0x00000000017b0648 in predelete_handler (p_object=0x5549540) at core/object.cpp:1781
#80 0x000000000078200f in memdelete<Node> (p_class=0x5549540) at core/os/memory.h:277
#81 0x0000000000dd04b6 in Node::_notification (this=0x5548160, p_notification=1) at scene/main/node.cpp:131
#82 0x000000000045c4c4 in Node::_notificationv (this=0x5548160, p_notification=1, p_reversed=true) at ./scene/main/node.h:44
#83 0x000000000045cac7 in CanvasItem::_notificationv (this=0x5548160, p_notification=1, p_reversed=true) at ./scene/2d/canvas_item.h:87
#84 0x000000000045d09f in Control::_notificationv (this=0x5548160, p_notification=1, p_reversed=true) at ./scene/gui/control.h:50
#85 0x0000000000565fe3 in Container::_notificationv (this=0x5548160, p_notification=1, p_reversed=true) at ./scene/gui/container.h:37
#86 0x00000000005665bb in BoxContainer::_notificationv (this=0x5548160, p_notification=1, p_reversed=true) at ./scene/gui/box_container.h:37
#87 0x00000000005674df in VBoxContainer::_notificationv (this=0x5548160, p_notification=1, p_reversed=true) at ./scene/gui/box_container.h:80
#88 0x00000000017a861a in Object::notification (this=0x5548160, p_notification=1, p_reversed=true) at core/object.cpp:863
#89 0x00000000017a609f in Object::_predelete (this=0x5548160) at core/object.cpp:259
---Type <return> to continue, or q <return> to quit---
#90 0x00000000017b0648 in predelete_handler (p_object=0x5548160) at core/object.cpp:1781
#91 0x000000000078200f in memdelete<Node> (p_class=0x5548160) at core/os/memory.h:277
#92 0x0000000000dd04b6 in Node::_notification (this=0x4029890, p_notification=1) at scene/main/node.cpp:131
#93 0x000000000045c4c4 in Node::_notificationv (this=0x4029890, p_notification=1, p_reversed=true) at ./scene/main/node.h:44
#94 0x000000000045cac7 in CanvasItem::_notificationv (this=0x4029890, p_notification=1, p_reversed=true) at ./scene/2d/canvas_item.h:87
#95 0x000000000045d09f in Control::_notificationv (this=0x4029890, p_notification=1, p_reversed=true) at ./scene/gui/control.h:50
#96 0x000000000084d1ad in Panel::_notificationv (this=0x4029890, p_notification=1, p_reversed=true) at ./scene/gui/panel.h:39
#97 0x00000000017a861a in Object::notification (this=0x4029890, p_notification=1, p_reversed=true) at core/object.cpp:863
#98 0x00000000017a609f in Object::_predelete (this=0x4029890) at core/object.cpp:259
#99 0x00000000017b0648 in predelete_handler (p_object=0x4029890) at core/object.cpp:1781
#100 0x000000000078200f in memdelete<Node> (p_class=0x4029890) at core/os/memory.h:277
#101 0x0000000000dd04b6 in Node::_notification (this=0x4029290, p_notification=1) at scene/main/node.cpp:131
#102 0x000000000045c4c4 in Node::_notificationv (this=0x4029290, p_notification=1, p_reversed=true) at ./scene/main/node.h:44
#103 0x000000000045cac7 in CanvasItem::_notificationv (this=0x4029290, p_notification=1, p_reversed=true) at ./scene/2d/canvas_item.h:87
#104 0x000000000045d09f in Control::_notificationv (this=0x4029290, p_notification=1, p_reversed=true) at ./scene/gui/control.h:50
#105 0x00000000017a861a in Object::notification (this=0x4029290, p_notification=1, p_reversed=true) at core/object.cpp:863
#106 0x00000000017a609f in Object::_predelete (this=0x4029290) at core/object.cpp:259
#107 0x00000000017b0648 in predelete_handler (p_object=0x4029290) at core/object.cpp:1781
#108 0x000000000078200f in memdelete<Node> (p_class=0x4029290) at core/os/memory.h:277
#109 0x0000000000dd04b6 in Node::_notification (this=0x370bf00, p_notification=1) at scene/main/node.cpp:131
#110 0x000000000045c4c4 in Node::_notificationv (this=0x370bf00, p_notification=1, p_reversed=true) at ./scene/main/node.h:44
#111 0x00000000008147a3 in EditorNode::_notificationv (this=0x370bf00, p_notification=1, p_reversed=true) at editor/editor_node.h:99
#112 0x00000000017a861a in Object::notification (this=0x370bf00, p_notification=1, p_reversed=true) at core/object.cpp:863
#113 0x00000000017a609f in Object::_predelete (this=0x370bf00) at core/object.cpp:259
#114 0x00000000017b0648 in predelete_handler (p_object=0x370bf00) at core/object.cpp:1781
#115 0x000000000078200f in memdelete<Node> (p_class=0x370bf00) at core/os/memory.h:277
#116 0x0000000000dd04b6 in Node::_notification (this=0x366f7d0, p_notification=1) at scene/main/node.cpp:131
#117 0x000000000045c4c4 in Node::_notificationv (this=0x366f7d0, p_notification=1, p_reversed=true) at ./scene/main/node.h:44
#118 0x0000000000e148b5 in Viewport::_notificationv (this=0x366f7d0, p_notification=1, p_reversed=true) at scene/main/viewport.h:76
#119 0x00000000017a861a in Object::notification (this=0x366f7d0, p_notification=1, p_reversed=true) at core/object.cpp:863
---Type <return> to continue, or q <return> to quit---
#120 0x00000000017a609f in Object::_predelete (this=0x366f7d0) at core/object.cpp:259
#121 0x00000000017b0648 in predelete_handler (p_object=0x366f7d0) at core/object.cpp:1781
#122 0x0000000000df847e in memdelete<Viewport> (p_class=0x366f7d0) at core/os/memory.h:277
#123 0x0000000000defa78 in SceneTree::finish (this=0x3672010) at scene/main/scene_main_loop.cpp:546
#124 0x0000000000412bff in OS_X11::run (this=0x7fffffffd4f0) at platform/x11/os_x11.cpp:1916
#125 0x000000000040b07f in main (argc=2, argv=0x7fffffffda48) at platform/x11/godot_x11.cpp:49
```

Likely a bug introduced by #8979, ping @Geequlim.

**Steps to reproduce:**
- Start the Godot editor
- Close it

**Comment by RandomShaper** [comment]
Those 0xEA should raise a red flag about dangling pointers.

In this case it seems that the PropertyEditor was already freed and therefore a pointer member from it points to 0xeaea...

That's why I'm trying to add byte tagging to 3.0 and in a more complete way, so by looking at crash addresses we can infer whether the engine is trying to access already freed memory or allocated but uninitialized data (both tagged differently).

**Comment by akien-mga** [comment]
Fixed by #9655.
